tests/engine/effect_test.py

```diff
@@ -0,0 +1,90 @@
+import pytest
+from tests.mocks import world_mock
+from scripts.engine import world
+from scripts.engine.core.constants import AfflictionTrigger
+from scripts.nqp.actions.afflictions import Affliction
+from scripts.engine.component import Afflictions, Identity, Position, Knowledge
+from scripts.engine.effect import TriggerAfflictionsEffect, ReduceSkillCooldownEffect
+from scripts.nqp.actions.skills import Move
+
+
+class MockAffliction(Affliction):
```

Comment:
We setup some default classes that represent real afflictions with different triggers. This is called [mocking](https://stackoverflow.com/questions/2665812/what-is-mocking). In short:

> Mocking is primarily used in unit testing. An object under test may have dependencies on other (complex) objects. To isolate the behavior of the object you want to replace the other objects by mocks that simulate the behavior of the real objects. This is useful if the real objects are impractical to incorporate into the unit test.

This can be either done manually using inheritance and a few tricks or you can use a library like [unittest.mock](https://docs.python.org/dev/library/unittest.mock.html) (It's included in Python). I added some manual examples since the library is easier to use.